### ISSUE TITLE
Subscribing to query params not working after returning to the page

### ISSUE THREAD

**Issue by ildar-icoosoft** [ionic-team/ionic#16582]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
Ionic:

   ionic (Ionic CLI)             : 4.3.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.12
   @angular-devkit/build-angular : 0.7.5
   @angular-devkit/schematics    : 0.7.5
   @angular/cli                  : 6.1.5
   @ionic/angular-toolkit        : 1.1.0

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : none
   Cordova Plugins       : no whitelisted plugins (0 plugins total)

System:

   Android SDK Tools : 26.1.1 (/Users/ildar/Library/Android/sdk)
   ios-deploy        : 1.9.2
   NodeJS            : v11.0.0 (/usr/local/bin/node)
   npm               : 6.4.1
   OS                : macOS High Sierra
   Xcode             : Xcode 9.4 Build version 9F1027a

```

**Describe the Bug**
I have a page: /news with query parameter page.

I have a subscription to query params:
```
ngOnInit() {
    this.route.queryParams.subscribe(params => {
        ...
    });
}
```
If I open /news and navigate to /news?page=1 then subscription works fine.

But if I open /news and navigate to /another-page and then again navigate to /news and navigate to /news?page=1 then subscription is not working anymore.

**Comment by bryce13950** [comment]
If you change the page id does the subscription callback still get hit?

**Comment by ildar-icoosoft** [comment]
@bryce13950 
Here is my another one bug about it: https://github.com/ionic-team/ionic/issues/16581
If I change query parameter "page" (this.router.navigate(['/news'], {queryParams: {page: 2}} ) ) then only subscription callback is executed. If I change parameter "page" (this.router.navigate(['/news', {page: 3} ]) ) then whole component is reinitialized.

**Comment by bryce13950** [comment]
OK, so is it changing when it is reinitialized versus simply hitting the subscription randomly?

**Comment by ildar-icoosoft** [comment]
If I am on page /news, and I subscribe to params ( this.route.params.subscribe(...) ), and I change parameter ( router.navigate(['/news', {page: 2}]) ) then:
1. new instance of NewsPage is created (constructor, ngOnInit, etc are executed)
2. old NewsPage instance is NOT destroyed (ngOnDestroy is NOT executed)
3. Query params subscription in old NewsPage is NOT executed

**Comment by ildar-icoosoft** [comment]
Now I tried to use latest Ionic (v4.0.0-beta.16) and it works much better than v4.0.0-beta.12 from my issue. Here is sample app: https://github.com/ildar-icoosoft/ionic-sample-route-bug
I found only 2 errors:

1. If I change query params nothing happens. Subscription callback is not executed. Component is not reinitialized.
2. If I change params then component is reinitialized (destroyed and created)

Both errors are fixed if I will remove line { provide: RouteReuseStrategy, useClass: IonicRouteStrategy } from AppModule
